cloudferrylib/os/compute/server_groups.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2014 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+"""
+This module contains the logic for collecting server group information from a
+cloud and deploying server groups on a cloud.
+
+Example:
+    src_handler = Handler(src_cloud)
+    dst_handler = Handler(dst_cloud)
+    dst_handler.deploy_server_groups(src_handler.get_server_groups())
+"""
+
+import pprint
+
+from novaclient import exceptions as nova_exc
+
+from cloudferrylib.base.resource import Resource
+from cloudferrylib.utils import utils as utl
+
+
+LOG = utl.get_log(__name__)
+
+SQL_SELECT_ALL_GROUPS = ("SELECT user_id, project_id, uuid, name, id FROM "
+                         "instance_groups WHERE deleted=0;")
+SQL_SELECT_POLICY = ("SELECT policy FROM instance_group_policy WHERE "
+                     "group_id=%s AND deleted=0;")
+SQL_SELECT_GROUP_ID = "SELECT id FROM instance_groups WHERE uuid='{0}';"
+SQL_DELETE_MEMBER = "DELETE FROM instance_group_member WHERE group_id={0};"
+SQL_DELETE_POLICY = "DELETE FROM instance_group_member WHERE group_id={0};"
+SQL_DELETE_GROUP = "DELETE_FROM instance_groups WHERE uuid='{0}';"
+SQL_INSERT_GROUP = ("INSERT INTO instance_groups (uuid, name, project_id, "
+                    "user_id, deleted) VALUES('{0}', '{1}', '{2}', '{3}', 0);")
+SQL_INSERT_POLICY = ("INSERT INTO instance_group_policy (group_id, policy, "
+                     "deleted) VALUES({0}, '{1}', 0)")
```

Comment:
I'm still getting familiar with the existing cloud ferry code base. I think I can now do these with some of the existing utility functions and the APIs.